app/components/admin/settings/row_component.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::Settings::RowComponent < ApplicationComponent
+  attr_reader :key, :tab, :type
+  delegate :dom_id, to: :helpers
+
+  def initialize(key, type: :text, tab: nil)
+    @key = key
+    @type = type
+    @tab = tab
+  end
+
+  def setting
+    @setting ||= Setting.find_by(key: key)
```

Comment:
While testing the map markers feature, I got an error:

```
NoMethodError in Admin::Settings#index
undefined method `to_key' for nil:NilClass
      key = convert_to_model(record).to_key

Extracted source (around line #3): 

class Admin::Settings::RowComponent < ApplicationComponent
  attr_reader :key, :tab, :type
  delegate :dom_id, to: :helpers
```

It took me a while to realize that I was getting the error because the setting wasn't present in my local database, and so the `dom_id` method fails.

So, since the application crashes if the setting isn't found, we could write:

```suggestion
    @setting ||= Setting.find_by!(key: key)
```

This way we get:

```
ActiveRecord::RecordNotFound in Admin::Settings#index
Couldn't find Setting

Extracted source (around line #12): 

  def setting
    @setting ||= Setting.find_by!(key: key)
  end
```

Which IMHO is more intuitive.

What do you think?

Reply:
Much better! 👌🏼